datarepo-clienttests/src/main/java/scripts/testscripts/RetrieveSnapshot.java

```diff
@@ -34,14 +41,54 @@ public RetrieveSnapshot() {
 
   private SnapshotSummaryModel snapshotSummaryModel;
   private List<BlobId> scratchFiles = new ArrayList<>();
+  private boolean deleteProfile;
 
   public void setup(List<TestUserSpecification> testUsers) throws Exception {
-    // create the profile and dataset
-    super.setup(testUsers);
+    // Specify the name of a profile to use for the test with the TEST_RUNNER_BILLING_PROFILE_NAME
+    // env variable.
+    // If it's not specified or the profile isn't found, a new one with a randomly generated name
+    // will be created
+    String profileNameRaw = System.getenv("TEST_RUNNER_BILLING_PROFILE_NAME");
+    Optional<String> profileName;
+    if (StringUtils.isBlank(profileNameRaw)) {
+      profileName = Optional.empty();
+      deleteProfile = true;
+      logger.info("No profile name specified.  Will create a new billing profile");
+    } else {
+      profileName = Optional.ofNullable(profileNameRaw);
+      deleteProfile = false;
+      logger.info("Will attempt to use a billing profile named {}", profileName.get());
+    }
+
+    // pick the a user that is a Data Repo steward to be the dataset creator
+    datasetCreator = SAMUtils.findTestUserThatIsDataRepoSteward(testUsers, server);
 
     // get the ApiClient for the snapshot creator, same as the dataset creator
     ApiClient datasetCreatorClient = DataRepoUtils.getClientForTestUser(datasetCreator, server);
     RepositoryApi repositoryApi = new RepositoryApi(datasetCreatorClient);
+    ResourcesApi resourcesApi = new ResourcesApi(datasetCreatorClient);
+    BillingProfileModel billingProfileModel =
+        profileName
+            .map(
+                p -> {
+                  try {
+                    List<BillingProfileModel> profiles =
+                        resourcesApi.enumerateProfiles(0, 100).getItems();
+
+                    BillingProfileModel foundProfile =
+                        IterableUtils.find(
+                            profiles, m -> StringUtils.equalsIgnoreCase(p, m.getProfileName()));
+                    if (foundProfile == null) {
+                      logger.info("Could not find a profile named {}", p);
+                    }
+                    return foundProfile;
```

Comment:
This can be done with Java APIs only
```suggestion
      Optional<BillingProfileModel> first = resourcesApi.enumerateProfiles(0, 100).getItems()
          .stream().filter(m -> profileName.equalsIgnoreCase(m.getProfileName()))
          .findFirst();
```
Personally I would use `orElseThrow()` on the `Optional` since referring to a unknown profile name seems like a fatal error.

Reply:
I was torn on that.  And I guess the stream would be backed by a list so it doesn't need to be explicitly closed